src/ClusterManager/job_manager.py

```diff
@@ -720,20 +735,119 @@ def mark_schedulable_preemptable_jobs(jobs_info, cluster_schedulable):
                     "required job resource %s.", job_id, cluster_schedulable,
                     job_resource)
 
+# schedule non-preempt part of queued inference jobs
+def mark_schedulable_inference_jobs_non_preemptable_part(jobs_info, cluster_schedulable, vc_schedulables):
+    for job_info in jobs_info:
+        if job_info["jobtrainingtype"] != "InferenceJob" or job_info["status"] != "queued":
+            continue
+
+        job_resource = job_info["job_resource"]
+        job_id = job_info["jobId"]
+        vc_name = job_info["job"]["vcName"]
+        vc_schedulable = vc_schedulables.get(vc_name)
+        if vc_schedulable is None:
+            logger.warning(
+                "vc %s is not exist as provided by %s, ignore this job",
+                vc_name, job_id)
+            continue
+
+        if cluster_schedulable >= job_resource and \
+                vc_schedulable >= job_resource:
+            vc_schedulable -= job_resource
+            cluster_schedulable -= job_resource
+            job_info["allowed"] = True
+            job_info["allowed_resource"] = copy.deepcopy(job_resource)
+            logger.info(
+                "Allow inferece job %s non-preemptable part from %s to run, job resouce %s",
+                job_id, vc_name, job_resource)
+        else:
+            logger.info(
+                "Disallow inference job %s non-preemptable part from vc %s to run."
+                "Requiring %s, vc_schedulable %s, cluster_schedulable %s",
+                job_id, vc_name, job_resource, vc_schedulable,
+                cluster_schedulable)
+
+# schedule preempt part of running/scheduling/queued inference jobs
+def mark_schedulable_inference_jobs_preemptable_part(jobs_info, cluster_schedulable):
+    empty_resource = ClusterResource()
+    for job_info in jobs_info:
+        # skip training job, or
+        # inference job which non-preemptable part is not allowed or has no preemptable part
+        if job_info["jobtrainingtype"] != "InferenceJob" or \
+            job_info["allowed"] is False or \
+            job_info["job_preemptable_resource"] is None:
+            continue
+
+        job_preemptable_resource = job_info["job_preemptable_resource"]
+        job_id = job_info["jobId"]
+
+        if cluster_schedulable != empty_resource:
```

Comment:
Because we want to allocate resource as long as cluster resource is not empty, no matter cluster resource is sufficient or not.  If cluster_resource is sufficient, we will allocate maxgpu. If not, we will allocate cluster's remaining gpu.

Reply:
I think this will never be False. For example some nodes have some free CPU.

Reply:
`cluster_schedulable` will hardly be empty. E.g.
```
"gpu": {"Standard_ND24rs": 0},
"cpu": {"Standard_ND24rs": 100},
"memory": {"Standard_ND24rs": "100Gi"},
```
In this case, the `schedulable_resource` here becomes 0. Then the logging at 808 doesn't sound appropriate, because no preemptable part has been admitted for scheduling.

Reply:
Got it, I will resolve it